### DISCUSSION TITLE
And/Or Operators

### DISCUSSION THREAD

**Discussion by jadamconnor** [wp-graphql/wp-graphql#2186] [General]
Is it possible to use them? Specifically with where... Something like:

```
posts(where: {_and: [{categoryName: "foo"}, {categoryName: bar}]}) {
    edges {
      node {
        id
      }
    }
  }
```

If not, can anyone think of a workaround that achieves the same thing?

I'm thinking I could do this:

```
posts(where: {categoryName: "foo"}) {
    edges {
      node {
        id
        categories(where: {name: ["bar"]}) {
          edges {
            node {
              id
            }
          }
        }
      }
    }
  }
```

And then some operation on the returned data that filters out posts with falsy categories objects?

I can see a potential downside being we're fetching data we don't need.

**Comment by jadamconnor** [comment]
Nothing? Could it be that no one here knows the answer to this?

**Comment by chriszarate** [comment]
You can query for the categories and then connect posts (OR operator):

```gql
query PostsByCategories {
  categories(
    where: {
      name: ["category1", "category2"]
    }
  ) {
    nodes {
      id
      name
      posts {
        nodes {
          id
          title
        }
      }
    }
  }
}
```

**Reply by jadamconnor** [reply]
Thanks for the response. This is more of a workaround of course, but useful. Still, it returns a whole lot of data we don't need, so not optimal.
